### DISCUSSION TITLE
Overriding the LIMIT clause for snowflake

### DISCUSSION THREAD

**Discussion by carrasquel** [sqlalchemy/sqlalchemy#10956] [Usage Questions]
Greetings, we are using the `snowflake-connector-python` and `snowflake-sqlalchemy` to connect to snowflake database using sqlalchemy, but we also connect to other databases, in some queries we can use `OFFSET` without the `LIMIT`, but in order for this to work with snowflake the limit clause has to be set with a null value `LIMIT NULL OFFSET X`. sqlalchemy for this case where there is no limit defined but there is offset, sets limit to `-1`. [sql compiler](https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4/lib/sqlalchemy/sql/compiler.py#L3889-L3897)
For snowflake this is invalid, how can I override the limit clause but only to work with snowflake extension?

**Comment by CaselIT** [comment]
Hi,

This looks like a bug of snowflake that you should report to the maintainers of the sqlalchemy extension.
>  For snowflake this is invalid, how can I override the limit clause but only to work with snowflake extension?

An engine or a connection has a `dialect` property. From it you can check the name of the driver in use and decide to do different things while creating the query
